contracts/masset/liquidator/Liquidator.sol

```diff
@@ -0,0 +1,303 @@
+pragma solidity 0.5.16;
+pragma experimental ABIEncoderV2;
+
+// External
+import { IUniswapV2Router02 } from '../platform-integrations/IUniswapV2Router02.sol';
+import { ICERC20 } from '../platform-integrations/ICompound.sol';
+import { IPlatformIntegration } from '../../interfaces/IPlatformIntegration.sol';
+// import { ILiquidator } from '../../interfaces/ILiquidator.sol';
+
+// Internal
+import { Initializable } from "@openzeppelin/upgrades/contracts/Initializable.sol";
+import { InitializableModule } from "../../shared/InitializableModule.sol";
+import { InitializableReentrancyGuard } from "../../shared/InitializableReentrancyGuard.sol";
+import { InitializableAbstractIntegration } from "../platform-integrations/InitializableAbstractIntegration.sol";
+
+// Libs
+import { SafeMath } from "@openzeppelin/contracts/math/SafeMath.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+interface Integration {
+    
+}
+
+/**
+ * @title   Liquidator
+ * @author  Stability Labs Pty. Ltd.
+ * @notice  The Liquidator allows rewards to be swapped for another token
+ *          and returned to a calling contract
+ * @dev     VERSION: 1.0
+ *          DATE:    2020-09-17
+ */
+contract Liquidator is
+    Initializable,
+    InitializableModule,
+    InitializableReentrancyGuard
+    {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    event LiquidationCreated(address indexed bAsset);
+    event LiquidationDeleted(address indexed bAsset);
+    event LiquidationUpdated(address indexed bAsset);
+    event LiquidationTriggered(address indexed bAsset);
+    event UniswapUpdated(address indexed uniswapAddress);
+
+    address public uniswapAddress;
+
+    struct Liquidation {
+        address     bAsset;
+        address     integration;
+        address     sellToken;
+        address[]   uniswapPath;
+        uint        amount;
+        bool        paused;
+        uint        lastTriggered;
+    }
+
+    mapping(address => Liquidation) public liquidations;
```

Comment:
Given that there is only 1 liquidation per `Integration`, the mapping should be `integration` centric and not `bAsset` centric. Should probably change this. After which, there is no need to store the actual integration address in the struct - that is double saving.

Reply:
The `Integration` interface doesn't have a way to read the `bAsset` and a lot of the logic is around the underlying `bAsset` so I favour leaving it this way. Plus if the integration contract changes for a bAsset this is supported (e.g. using Aave instead of Compound for DAI).